agent/taskresource/volume/dockervolume.go

```diff
@@ -76,40 +102,124 @@ func NewVolumeResource(name string,
 		client: client,
 		ctx:    ctx,
 	}
+	v.initStatusToTransitions()
+	return v, nil
+}
+
+func (vol *VolumeResource) Initialize(resourceFields *taskresource.ResourceFields) {
+	vol.ctx = resourceFields.Ctx
+	vol.client = resourceFields.DockerClient
+	vol.initStatusToTransitions()
+}
+
+func (vol *VolumeResource) initStatusToTransitions() {
+	statusToTransitions := map[resourcestatus.ResourceStatus]func() error{
+		resourcestatus.ResourceStatus(VolumeCreated): vol.Create,
+	}
+
+	statusToTransitions[resourcestatus.ResourceStatus(VolumeRemoved)] = vol.Cleanup
```

Comment:
`Cleanup` need not be added to the status to transitions map since it is not called during `progressTask`. Resources cleanup happen here: 
https://github.com/aws/amazon-ecs-agent/blob/8dae73394282df16570d41c067fa364652db4e31/agent/engine/docker_task_engine.go#L464

Reply:
Yes, but it shouldn't hurt to add it here. As the `REMOVED` should match to the `Cleanup` in the transition function.

Reply:
Since it is not being used anywhere, it would be preferable to remove this.